torch/_dynamo/eval_frame.py

```diff
@@ -311,6 +311,15 @@ def get_compiler_fn(compiler_fn):
 def lookup_backend(compiler_fn):
     """Expand backend strings to functions"""
     if compiler_fn == "inductor":
+        if (
+            torch.backends.cuda.matmul.allow_tf32 is False
+            and torch.cuda.get_device_capability() >= (7, 0)
+        ):
+            warnings.warn(
+                "Tensor cores are available but not enabled."
+                "Consider setting torch.backends.cuda.matmul.allow_tf32 == True in your python script for speedups"
```

Comment:
Nit: maybe we should use get/set_matmul_precision instead https://pytorch.org/docs/stable/generated/torch.set_float32_matmul_precision.html?highlight=matmul+precision#torch.set_float32_matmul_precision

Reply:
I've seen other libraries just go ahead and set the matmul precision but should we make this choice on behalf of users and modify their code? Not sure what's the precedent for these kinds of decisions

Reply:
If we are certain that inductor is used only for dl workloads, then we should, but I'm not certain of that.

Reply:
More of a fun project by @srush https://github.com/srush/torch-golf might have issues with this switch. More generally I guess I worry more about the hard science community that happens to be using pytorch although I'm not sure how big it actually is

Reply:
TBH I think this is one of those things that's best to monitor based on what users do, so I'll monitor issues people open but otherwise I think the current warning is a good compromise

Reply:
Yeah I agree we shouldn't switch the default, it will be too confusing, and we already know that enabling it by default causes a lot of issues, given how hard those issues are to debug, even if most people end up setting tf32, default should still be off. I still suggest using `torch.set_float32_matmul_precision` in the warning message.